src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarReader.cs

```diff
@@ -54,8 +54,18 @@ public TarReader(Stream archiveStream, bool leaveOpen = false)
         /// <remarks>The <see cref="TarEntry.DataStream"/> property of any entry can be replaced with a new stream. If the user decides to replace it on a <see cref="TarEntry"/> instance that was obtained using a <see cref="TarReader"/>, the underlying stream gets disposed immediately, freeing the <see cref="TarReader"/> of origin from the responsibility of having to dispose it.</remarks>
         public void Dispose()
         {
-            Dispose(disposing: true);
-            GC.SuppressFinalize(this);
```

Comment:
I implemented the Dispose/DisposeAsync based on our article: https://docs.microsoft.com/en-us/dotnet/standard/garbage-collection/implementing-dispose

If it's not already written down, we could consider expanding the article to describe cases where the private Dispose method isn't needed.

Reply:
> I implemented the Dispose/DisposeAsync based on our article

These types are sealed and don't declare finalizers.